packages/fx-core/src/common/spec-parser/utils.ts

```diff
@@ -129,16 +129,27 @@ export function isSupportedApi(
   method: string,
   path: string,
   spec: OpenAPIV3.Document,
-  allowMissingId: boolean
+  allowMissingId: boolean,
+  allowAPIKeyAuth: boolean
 ): boolean {
   const pathObj = spec.paths[path];
   method = method.toLocaleLowerCase();
   if (pathObj) {
     if (
       (method === ConstantString.PostMethod || method === ConstantString.GetMethod) &&
-      pathObj[method] &&
-      !pathObj[method]?.security
+      pathObj[method]
     ) {
+      const securities = pathObj[method]!.security;
+      const apiKeyAuth = getAPIKeyAuth(securities, spec);
+
+      if (!allowAPIKeyAuth && securities) {
```

Comment:
The logic may have problem later when more securities type supported. What we want to do is only security type equal to apiKey right?

Reply:
Yes, in the future if we support aad, the logic would be update to:
if (!allowAPIKeyAuth && !allowAadAuth && securities) {
   return false;
}